### PR TITLE
Added authorization on route for asset creation

### LINKED ISSUE #7081: Registering assets with state admin read-only account
----Describe the bug-----

State Admin Read-only users are restricted from registering an asset. However, if we write the complete URL (.../assets/new), we can register an asset successfully.

----Steps to Reproduce----
1- Log in using provided credentials
2- Click https://care.coronasafe.in/facility/497fd4fc-c5c1-4ed3-9b57-64cc881087fe/assets/new
3- Fill the asset details
4- Click on "Create Asset"

----Login Credentials----
Username: state_read123
Password: Lilo@123

----Screenshots----

### PR DESCRIPTION
**Pull request by kshitijv256**
## Proposed Changes

- Fixes #7081 
- Added Authorized route on asset creation allowing only NonReadonly user to access

### Related backend PR

- https://github.com/coronasafe/care/pull/1868

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

### CODE REVIEW

File: src/Utils/AuthorizeFor.tsx [hunk 1]
```diff
@@ -27,6 +27,10 @@ export type AuthorizedElementProps = {
   authorizeFor?: AuthorizedForCB | undefined;
 };
 
+export const NonReadOnlyUsersList = USER_TYPES.filter((userType) => {
+  return !userType.includes("ReadOnly");
+});
```
Comment (rithviknishad) [inline src/Utils/AuthorizeFor.tsx]: We already have the filter function defined just below it. Re-use it.

```suggestion
export const NonReadOnlyUsersList = USER_TYPES.filter(NonReadOnlyUsers);
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]
@kshitijv256 Update permissions in backend too to disallow this operation for read only users

**Review by rithviknishad — CHANGES_REQUESTED** [review]
by mistake

**Review by rithviknishad — APPROVED** [review]

**Comment by rithviknishad** [comment]
@kshitijv256 still not followed up on the backend PR
https://github.com/coronasafe/care/pull/1868#pullrequestreview-1896428501

**Comment by kshitijv256** [comment]
@rithviknishad I have made the requested changes in backed PR as well, can I take on other issues or wait for that to be closed first ?

**Comment by rithviknishad** [comment]
Yes @kshitijv256 you can take other issues 🚀

**Pull request closed without merging**